tacs/problems/static.py

```diff
@@ -101,6 +118,8 @@ class StaticProblem(TACSProblem):
             "Flag for printing out timing information for class procedures.",
         ],
     }
+    defaultOptions.update(tacs.solvers.NewtonSolver.defaultOptions)
+    defaultOptions.update(tacs.solvers.ContinuationSolver.defaultOptions)
```

Comment:
This adds a lot of options to the StaticProblem class that are specific to the nonlinear solver. Can we reduce the number of nonlinear solver options to 3 or 4 (that basic users would use) and move the rest to the NLSolver classes?

If some of these options are still needed by power users we can have them live in the solver classes and allow the user to modify them directly?

Reply:
My intention here was to allow the user to set the nonlinear options through the problem interface directly, without having to know anything about the `nonlinearSolver` object that the problem holds. I would prefer not to split the options up between the problem and solver classes. IMO the user should either set all solver options through the problem or all directly through the solver, that way they don't need to remember which options are set where.

Here I was adding the nonlinear solver default options so that the static problem's `setOption` method wouldn't complain when the user sets a nonlinear solver option. I think I can rejig this implementation so that the user can still set the nonlinear solver options through the problem without adding these default options, that way we don't blow up the static problem options documentation. Would that work for you?

Reply:
I agree with your first point here. To avoid confusion, solver options should be owned by the solvers (not the problems). 

If the user wants to update an option for the nonlinear solver the API could look something like this:
```
problem.nonlinearSolver.setOption("relTol", 1e-8)
```

Additionally, this allows us to get rid of the awkward convention of including the solver name in each of its options (i.e. `newtonRelTol`->`relTol`).

Ideally, the only linear/nonlinear solver options that will be set at the problem level is which solver class to use for each. This means eventually we'll have to move the linear solver options to a separate class as well. We can do this in a separate PR, as it will likely require the creation of a wrapper class to hold the options for these solvers and come with some breaking/deprecating changes.

I think we can still begin prototyping this with just the nonlinear solvers in this PR if that sounds good to you @A-CGray.

Reply:
In the end I decided to just remove all the code that handles passing options from the problem to the solver and from the continuation solver to the newton solver. The user will have to pass options directly to the outer and inner nonlinear solvers, like [this](https://github.com/smdogroup/tacs/blob/215266839a9a68d747065850d28fef120960b30e/examples/nonlinear_cantilever/analysis.py#L85-L107).

I was initially against this approach because I didn't want the user to have to know about the hierarchy of the nonlinear solvers. But now that I think about it, they would need to have a basic understanding of that to know which options they want to set anyway.